api/integrations/github/dataclasses.py

```diff
@@ -1,44 +1,54 @@
-import typing
-from dataclasses import dataclass
-from typing import Optional
+from abc import ABC
+from dataclasses import dataclass, field
+from typing import Any, Optional
 
 
+# Base Dataclasses
 @dataclass
 class GithubData:
     installation_id: str
     feature_id: int
     feature_name: str
     type: str
-    feature_states: list[dict[str, typing.Any]] | None = None
+    feature_states: list[dict[str, Any]] | None = None
     url: str | None = None
     project_id: int | None = None
     segment_name: str | None = None
 
     @classmethod
-    def from_dict(cls, data_dict: dict[str, typing.Any]) -> "GithubData":
+    def from_dict(cls, data_dict: dict[str, Any]) -> "GithubData":
         return cls(**data_dict)
 
 
 @dataclass
 class CallGithubData:
     event_type: str
     github_data: GithubData
-    feature_external_resources: list[dict[str, typing.Any]]
+    feature_external_resources: list[dict[str, Any]]
 
 
+# Dataclasses for external calls to GitHub API
 @dataclass
-class RepoQueryParams:
+class PaginatedQueryParams(ABC):
+    page: int = field(default=1, init=False)
+    page_size: int = field(default=100, init=False)

```

Comment:
1. Why derive from `ABC`?
2. Why `init=False`? What if we want to customise pagination?

Reply:
1. Because I forgot we need to use it for a request and thought we would only need it as a base class for others. Now I changed it to a regular class and it's in use by the get repos endpoint.
2. Because derivating without either `init=false` or `kw_only` gives an error. I now changed to use `kw_only` to have a better result.
Thanks for cathing this out!